src/Cli/func/Helpers/AzureHelper.cs

```diff
@@ -383,31 +383,61 @@ internal static async Task CheckFunctionHostStatusForFlex(Site functionApp, stri
             functionAppReadyClient.DefaultRequestHeaders.Add("User-Agent", Constants.CliUserAgent);
             functionAppReadyClient.DefaultRequestHeaders.Add("Accept", jsonContentType);
 
+            var statusUri = new Uri($"https://{functionApp.HostName}/admin/host/status?code={masterKey}");
+            ColoredConsole.WriteLine($"Target endpoint: {statusUri.GetLeftPart(UriPartial.Path)}");
+
+            int attemptCount = 1;
             await RetryHelper.Retry(
                 async () =>
                 {
+                    // Remove the header first if it already exists from a previous retry
+                    if (functionAppReadyClient.DefaultRequestHeaders.Contains("x-ms-request-id"))
+                    {
+                        functionAppReadyClient.DefaultRequestHeaders.Remove("x-ms-request-id");
+                    }
+
                     functionAppReadyClient.DefaultRequestHeaders.Add("x-ms-request-id", Guid.NewGuid().ToString());
-                    var uri = new Uri($"https://{functionApp.HostName}/admin/host/status?code={masterKey}");
+
                     var request = new HttpRequestMessage()
                     {
-                        RequestUri = uri,
-                        Method = HttpMethod.Get
+                        RequestUri = statusUri,
+                        Method = HttpMethod.Get,
                     };
 
-                    var response = await functionAppReadyClient.SendAsync(request);
-                    ColoredConsole.Write(".");
+                    ColoredConsole.WriteLine($"Attempt {attemptCount}: Calling admin/host/status endpoint...");
+
+                    using var cts = new CancellationTokenSource(TimeSpan.FromSeconds(30));
+                    var startTime = DateTime.UtcNow;
+
+                    var response = await functionAppReadyClient.SendAsync(request, cts.Token);
+
+                    var elapsed = DateTime.UtcNow - startTime;
+                    ColoredConsole.WriteLine($"Response received after {elapsed.TotalMilliseconds:F0}ms - Status: {response.StatusCode} ({(int)response.StatusCode})");
+
+                    var responseContent = await response.Content.ReadAsStringAsync();
```

Comment:
response content is only used in the else portion of the if-else below, can we move the retrieval of this content to that code path